packages/db/src/core/remove.ts

```diff
@@ -20,20 +20,29 @@ export async function remove<T, P extends keyof T>(
 ): Promise<boolean> {
   const primaryKeyProperty = table[BlinkKey].options.primary;
   const primaryKey = entity[primaryKeyProperty];
-  const deleted = table[BlinkKey].storage.primary.delete(primaryKey);
-  for (const [property, btree] of Object.entries<BTree<any, T[]>>(
+  const indexEntries = Object.entries<BTree<any, T[]>>(
     table[BlinkKey].storage.indexes as any
-  )) {
-    const key = (entity as any)[property];
-    if (key === null || key === undefined) continue;
+  );
+  // For tables without indexes, it is not necessary to retrieve the actual entity before deleting it
+  // For tables with indexes, we need the full entity to find all corresponding index entries
+  const item =
+    indexEntries.length > 0 ? table[BlinkKey].storage.primary.get(primaryKey) : null;
```

Comment:
I don't know if that's "too much optimization" in your opinion, but with this code, the "table with no indexes" case should not be slower with my fix.

Reply:
There can never be too much optimization :D